src/lualib-build/plugin.ts

```diff
@@ -0,0 +1,174 @@
+                const featureName = path.basename(moduleSpecifier.text, ".ts") as tstl.LuaLibFeature;
+                if (featureName in tstl.LuaLibFeature) {
+                    usedFeatures.add(featureName);
+                }
+            }
+        }
+
+        const filteredStatements = fileResult.statements
+            .filter(
+                s => !isExportTableDeclaration(s) && !isRequire(s) && !isImport(s, importNames) && !isExportsReturn(s)
+            )
+            .map(statement => {
+                if (isExportAlias(statement)) {
+                    const name = statement.left[0];
+                    const exportName = statement.right[0].index.value;
+                    if (name.text === exportName) return undefined; // Remove "x = x" statements
+                    return tstl.createAssignmentStatement(name, tstl.createIdentifier(exportName));
+                }
+                return statement;
+            })
+            .filter(statement => statement !== undefined) as tstl.Statement[];
+
+        const exportNames = filteredStatements.filter(isExportAssignment).map(s => s.left[0].index.value);
+        if (!filteredStatements.every(isExportAssignment)) {
+            // If there are local statements, wrap them in a do ... end with exports outside
+            const exports = tstl.createVariableDeclarationStatement(exportNames.map(k => tstl.createIdentifier(k)));
+            // transform export assignments to local assignments
+            const bodyStatements = filteredStatements.map(s =>
+                isExportAssignment(s)
+                    ? tstl.createAssignmentStatement(tstl.createIdentifier(s.left[0].index.value), s.right[0])
+                    : s
+            );
+
+            fileResult.statements = [exports, tstl.createDoStatement(bodyStatements)];
+        } else {
+            // transform export assignments to local variable declarations
+            fileResult.statements = filteredStatements.map(s =>
+                tstl.createVariableDeclarationStatement(tstl.createIdentifier(s.left[0].index.value), s.right[0])
+            );
+        }
+
+        // Save dependency information
+        this.featureExports.set(featureName, new Set(exportNames));
+        if (usedFeatures.size > 0) {
+            this.featureDependencies.set(featureName, usedFeatures);
+        }
+
+        return fileResult;
+    }
+
+    protected createLuaLibModulesInfo(): { result: LuaLibModulesInfo; diagnostics: ts.Diagnostic[] } {
+        const result: Partial<LuaLibModulesInfo> = {};
+        const diagnostics: ts.Diagnostic[] = [];
+        for (const feature of Object.values(tstl.LuaLibFeature)) {
+            const exports = this.featureExports.get(feature);
+            if (!exports) {
+                diagnostics.push(lualibDiagnostic(`Missing file for lualib feature: ${feature}`));
+                console.error(`Missing file for lualib feature: ${feature}`);
```

Comment:
Isn't the diagnostic enough?

Reply:
Yes, this looks like a commit merge oversight

Reply:
Removed